terra_notebook_utils/blobstore/url.py

```diff
@@ -0,0 +1,88 @@
+import os
+import shutil
+from math import ceil
+from functools import wraps
+from typing import Dict, Generator, Optional
+
+from requests.exceptions import HTTPError, ConnectionError
+from getm.reader import URLRawReader, URLReaderKeepAlive
+from getm.http import http
+from getm import checksum, default_chunk_size
+from getm.utils import indirect_open
+
+from terra_notebook_utils import blobstore
+
+
+def catch_blob_not_found(func):
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        try:
+            return func(self, *args, **kwargs)
+        except (HTTPError, ConnectionError) as ex:
+            raise blobstore.BlobNotFoundError(f"Could not find {self.url}") from ex
+    return wrapper
+
+class URLBlobStore(blobstore.BlobStore):
+    def blob(self, url: str) -> "URLBlob":
+        return URLBlob(url)
+
+class URLBlob(blobstore.Blob):
+    def __init__(self, url: str):
+        assert url.startswith("http://") or url.startswith("https://")
+        self.url = self.key = url
```

Comment:
I would not expect a variable named `self.key` to still contain the `schema`.

Reply:
Since http/s URLs are being abstracted as blobs, I don't have a problem with the key containing the schema.